### PR TITLE
feat(secret): support enable rules

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
The new field `enable-buitlin-rules` enables only specified rules. The detail is [here](https://github.com/aquasecurity/fanal/pull/488).

## Related PRs
- [x] https://github.com/aquasecurity/fanal/pull/488

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: docs/docs/secret/scanning.md [hunk 1]
```diff
@@ -92,9 +92,29 @@ If you don't need secret scanning, you can disable it via the `--security-checks
 $ trivy image --security-checks vuln alpine:3.15
 ```
 
+## Recommendation
+We would recommend specifying `--skip-dirs` for faster secret scanning.
+In container image scanning, Trivy walks the file tree rooted  `/` and scans all the files other than [built-in allowed paths][builtin].
+It will take a while if your image contains a lot of files even though Trivy tries to avoid scanning layers from a base image.
+If you want to make scanning faster, `--skip-dirs` and `--skip-files` helps so that Trivy will skip scanning those files and directories.
+The usage examples are [here][examples]
```
Comment (VaismanLior) [inline docs/docs/secret/scanning.md:100]:
```suggestion
The usage examples are [here][examples].
```

File: docs/docs/secret/scanning.md [hunk 2]
```diff
@@ -92,9 +92,29 @@ If you don't need secret scanning, you can disable it via the `--security-checks
 $ trivy image --security-checks vuln alpine:3.15
 ```
 
+## Recommendation
+We would recommend specifying `--skip-dirs` for faster secret scanning.
+In container image scanning, Trivy walks the file tree rooted  `/` and scans all the files other than [built-in allowed paths][builtin].
```
Comment (VaismanLior) [inline docs/docs/secret/scanning.md:97]: Would be easier to add a line number to the link, altho it will change frequently... 😢

Comment (knqyf263) [reply docs/docs/secret/scanning.md:97]: Yes, I'm worried that the reference will be stale soon. Then, users will refer to the old file. Any idea?

File: docs/docs/secret/examples.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# Examples
+
```
Comment (VaismanLior) [inline docs/docs/secret/examples.md:2]: Add a reference to QuickStart here?

**Review by VaismanLior — APPROVED** [review]

File: docs/docs/secret/scanning.md (hunk 2 above)
Comment (knqyf263) [reply docs/docs/secret/scanning.md]: I split builtin.go into builtin-rules.go and builtin-allow-rules.go.

**Pull request merged**